### PR TITLE
GPS search

### PR DESCRIPTION
**Pull request by thompson-zak**
:rocket: Ready

### CODE REVIEW

File: backend/search.py [hunk 1]
```diff
@@ -43,3 +44,49 @@ def search():
 
     results_json = results.exclude('image').to_json()
     return results_json
+
+
+@search_blueprint.route("/search_gps", methods=['GET'])
```
Comment (rsun0) [inline backend/search.py]: This belongs in its own file, search_gps.py. Each route/blueprint should be in its own file.

File: backend/tests/test_search.py [hunk 2]
```diff
@@ -128,3 +128,29 @@ def test_limit(client, db):
     response = json.loads(rv.data)
     assert len(response) == limit
     assert response[0]['upc'] == matches[0].upc
+
+
+def test_search_close(client, db):
```
Comment (rsun0) [inline backend/tests/test_search.py]: Please move these tests to their own file, test_gps_search.py

File: backend/tests/test_search.py [hunk 3]
```diff
@@ -128,3 +128,29 @@ def test_limit(client, db):
...
+    for i in test_data.valid_items:
+        copy.deepcopy(i).save()
+
+    lat = 0
+    long = 0
+    miles = 10
+
+    rv = client.get('/search_gps?lat=' + str(lat) + "&long=" + str(long) + "&miles=" + str(miles))
+    response = json.loads(rv.data)
+    assert len(response) == 2
```
Comment (rsun0) [inline backend/tests/test_search.py]: Please do not hardcode the test conditions; if we add more test stores, the number may change from 2. Instead, programmatically determine the correct number of stores to return (by iterating through the test items and checking their distance.)

File: backend/tests/test_search.py [hunk 4]
```diff
@@ -128,3 +128,29 @@ def test_limit(client, db):
...
+    for i in test_data.valid_items:
+        copy.deepcopy(i).save()
+
+    lat = 0
+    long = 0
+    miles = 1000
+
+    rv = client.get('/search_gps?lat=' + str(lat) + "&long=" + str(long) + "&miles=" + str(miles))
+    response = json.loads(rv.data)
+    assert len(response) == 3
```
Comment (rsun0) [inline backend/tests/test_search.py]: Please do not hardcode the test conditions; if we add more test stores, the number may change from 3. Instead, programmatically determine the correct number of stores to return (by iterating through the test items and checking their distance.)

File: backend/search.py [hunk 5]
```diff
@@ -43,3 +44,49 @@ def search():
...
+                'long': store_long
+            }
+            t = tuple(store_entry.items())
+            if t not in seen:
+                seen.add(t)
+                list_of_stores.append(store_entry)
+
+    list_of_stores_gps = []
+    for store in list_of_stores:
+        r = 3963.0
```
Comment (rsun0) [inline backend/search.py]: Please rename this to earth_radius so it's clear what it is.

**Review by rsun0 — CHANGES_REQUESTED** [review]
Please move your code into new files, make your tests more robust, and rename a variable.

File: backend/tests/test_search.py (hunk 2 above)
Comment (thompson-zak) [reply backend/tests/test_search.py]: Done

File: backend/tests/test_search.py (hunk 3 above)
Comment (thompson-zak) [reply backend/tests/test_search.py]: Done

File: backend/tests/test_search.py (hunk 4 above)
Comment (thompson-zak) [reply backend/tests/test_search.py]: Done

File: backend/search.py (hunk 5 above)
Comment (thompson-zak) [reply backend/search.py]: Done

File: backend/search.py (hunk 1 above)
Comment (thompson-zak) [reply backend/search.py]: Done

**Review by rsun0 — APPROVED** [review]

**Pull request merged**